cf-ops-automation-broker-core/src/test/java/com/orange/oss/cloudfoundry/broker/opsautomation/ondemandbroker/pipeline/CassandraProcessorTest.java

```diff
@@ -9,61 +9,199 @@
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.runners.MockitoJUnitRunner;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.cloud.servicebroker.model.CreateServiceInstanceRequest;
+import org.springframework.cloud.servicebroker.model.*;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import static junit.framework.TestCase.assertTrue;
+import static org.fest.assertions.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
 
 public class CassandraProcessorTest {
 
     private static final Logger logger = LoggerFactory.getLogger(CassandraProcessorTest.class);
     public static final String SERVICE_INSTANCE_ID = "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa0";
+    protected static final String TEMPLATES_REPOSITORY_ALIAS_NAME = "paas-template.";
+    protected static final String SECRETS_REPOSITORY_ALIAS_NAME = "paas-secrets.";
 
-    @Test
+    @Test@Ignore
     public void creates_structures_and_returns_response() {
-        //TODO
-        //Given
+        //Given a creation request
+        CreateServiceInstanceRequest creationRequest = new CreateServiceInstanceRequest("service_definition_id",
+                "plan_id",
+                "org_id",
+                "space_id",
+                null
+        );
+        creationRequest.withServiceInstanceId(SERVICE_INSTANCE_ID);
+
+        //Given a populated context
+        Context context = new Context();
+        context.contextKeys.put(ProcessorChainServiceInstanceService.CREATE_SERVICE_INSTANCE_REQUEST, creationRequest);
+        context.contextKeys.put(TEMPLATES_REPOSITORY_ALIAS_NAME + GitProcessorContext.workDir.toString(), aGitRepoWorkDir());
+        context.contextKeys.put(SECRETS_REPOSITORY_ALIAS_NAME + GitProcessorContext.workDir.toString(), aGitRepoWorkDir());
+
+        //Given a mock behaviour
```

Comment:
We have risks of having our test cases prevent us from changing the production code design.

See https://testing.googleblog.com/2017/12/testing-on-toilet-only-verify-state.html for background

Our mocking and assertions need to remain flexible enough to enable future refactoring